### PR TITLE
perf: lazy load toast-ui editor dependencies

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Dynamically load toast-ui editor languages and the syntax highlight plugin for markdown files. This improves editor loading times in other scenarios where of one these 2 things are not needed.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/10982

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-pkg/src/components/TextEditor.vue [hunk 1]
```diff
@@ -71,7 +50,59 @@ export default defineComponent({
...
+
+      return plugin.default
+    }
+
+    onMounted(async () => {
+      let codeSyntaxHighlight: unknown
+      if (!props.isReadOnly && unref(isMarkdown)) {
+        // only needed for md documents with write permissions
+        await loadTranslations()
+        codeSyntaxHighlight = await loadSyntaxHighlighting()
```
Comment (AlexAndBear) [inline packages/web-pkg/src/components/TextEditor.vue:100]: Syntax highlighting should be possible even if read only ? 🤷
Check space readme for example

**Review by AlexAndBear — CHANGES_REQUESTED** [review]

File: packages/web-pkg/src/components/TextEditor.vue (hunk 1 above)
Comment (JammingBen) [reply packages/web-pkg/src/components/TextEditor.vue:100]: Ah okay, I didn't know that 👍

Comment (AlexAndBear) [reply packages/web-pkg/src/components/TextEditor.vue:98]: I think that also applies to shared MD files with viewer permissions 👍

**Review by AlexAndBear — APPROVED** [review]

**Comment by AlexAndBear** [comment]
Looks a little clumsy but I fear that's the downside of making it performant 😁

**Comment by JammingBen** [comment]
> Looks a little clumsy but I fear that's the downside of making it performant 😁

Totally agree :( Also the weird dist-imports of the lib contribute to that unfortunately.

**Comment by kulmann** [comment]
> > Looks a little clumsy but I fear that's the downside of making it performant 😁
> 
> Totally agree :( Also the weird dist-imports of the lib contribute to that unfortunately.

Do you think this is something to solve or propose upstream? ;-)

**Comment by AlexAndBear** [comment]
Could you think of adding a new ts file for the translations where you import the lang files and then just importing that file Async, for now ?

**Comment by JammingBen** [comment]
> Do you think this is something to solve or propose upstream? ;-)

I didn't dig too deep, but it seems that you can use the non-dist imports, but then you have to explicitly name the supported languages for the syntax highlighting.

> > Could you think of adding a new ts file for the translations where you import the lang files and then just importing that file Async, for now ?
> 
> good idea 👍

Hmm I don't see much benefit tbh. Also it would raise the question where this file would live. We're not in an app context here, this is just a single `web-pkg` component. But if you 2 want to have it that way, sure that's possible 🤷

**Comment by AlexAndBear** [comment]
Just decluttering I guess. Could make a new Folder 
/Components/TextEditor/editor.vue
/Components/TextEditor/i18n.ts

**Pull request merged**